app/src/main/java/com/example/abha_create_verify_android/verify/MobileLinkedABHAActivity.kt

```diff
@@ -0,0 +1,216 @@
+import androidx.core.content.ContextCompat
+import androidx.core.content.res.ResourcesCompat
+import androidx.lifecycle.ViewModelProvider
+import com.example.abha_create_verify_android.AbhaAddressActivity
+import com.example.abha_create_verify_android.AbhaPatientProfileActivity
+import com.example.abha_create_verify_android.MainViewModel
+import com.example.abha_create_verify_android.PatientSubject
+import com.example.abha_create_verify_android.R
+import com.example.abha_create_verify_android.ViewModelFactory
+import com.example.abha_create_verify_android.data.api.ApiHelper
+import com.example.abha_create_verify_android.data.api.RetrofitBuilder
+import com.example.abha_create_verify_android.data.model.MobileNumberLinkedABHA
+import com.example.abha_create_verify_android.data.model.SearchAbhaReq
+import com.example.abha_create_verify_android.databinding.ActivityLinkedAbhaBinding
+import com.example.abha_create_verify_android.utils.DialogUtils
+import com.example.abha_create_verify_android.utils.Status
+import com.example.abha_create_verify_android.utils.Variables
+
+
+class MobileLinkedABHAActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityLinkedAbhaBinding
+    private lateinit var viewModel: MainViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLinkedAbhaBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setupViewModel()
+
+        setSupportActionBar(binding.appBarLayout.includeToolbar.toolbarAbha)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.title = resources.getString(R.string.verify_abha)
+
+        val dataList: ArrayList<MobileNumberLinkedABHA>? = intent.getSerializableExtra("linkedABHAs") as? ArrayList<MobileNumberLinkedABHA>
+
+        val listContainer = findViewById<LinearLayout>(R.id.listContainer)
+
+        var isABHAAddressLinked = false
+        var selectedIndex = -1
+        var selectedTextView: AppCompatTextView? = null
+
+        if (dataList != null) {
+            for ((index, dataItem) in dataList.withIndex()) {
+                val textView = createTextView(dataItem)
+
+                textView.setOnClickListener {
+                    binding.abhaAddressNotLinked.visibility = View.GONE
+                    binding.txtLinked.visibility = View.INVISIBLE
+
+                    selectedTextView?.setBackgroundColor(Color.WHITE)
+                    textView.setBackgroundResource(R.drawable.border)
+
+                    selectedTextView = textView
+                    selectedIndex = index
+
+                    if (dataList[selectedIndex].abhaAddress.isEmpty()) {
+                        binding.abhaAddressNotLinked.visibility = View.VISIBLE
+                        isABHAAddressLinked = false
+                    }
+                    else
+                    {
+                        isABHAAddressLinked = true
```

Comment:
instead of using else can we set the variable to true by default

Reply:
done